firecrest/BasicClient.py

```diff
@@ -611,14 +618,81 @@ def compress(self, machine: str, source_path: str, target_path: str) -> str:
             additional_headers={"X-Machine-Name": machine},
             data={"targetPath": target_path, "sourcePath": source_path},
         )
-        self._json_response([resp], 201)
+        timeout_str = "Command has finished with timeout signal"
+        if (
+            resp.status_code == 201 or
+            fail_on_timeout or
+            resp.status_code != 400 or
+            resp.json().get('error', '') != timeout_str
+        ):
+            self._json_response([resp], 201)
+        else:
+            logger.debug(
+                f"Compression of {source_path} to {target_path} has finished "
+                f"with timeout signal. Will submit a job to compress the "
+                f"file."
+            )
+            job_info = self.submit_compress_job(
+                machine,
+                source_path,
+                target_path
+            )
+            jobid = job_info['jobid']
+            active_jobs = self.poll_active(
+                machine,
+                [jobid]
+            )
+            intervals = itertools.cycle(
+                [1, 1, 1, 1, 1,
+                 5, 5, 5, 5,
+                 10, 10, 10,
+                 30]
+            )
+            while (
+                active_jobs and
+                not slurm_state_completed(active_jobs[0]['state'])
+            ):
+                time.sleep(next(intervals))
+                active_jobs = self.poll_active(
+                    machine,
+                    [jobid]
+                )
+
+            if (
+                active_jobs and
```

Comment:
same here, one could remove `active_jobs`. 
Infact this `if` will always be `False` since there is no `active_jobs` regardless of their state

```suggestion
```